src/causalprog/graph/graph.py

```diff
@@ -211,43 +209,59 @@ def roots_down_to_outcome(
             node for node in self.ordered_nodes if node == outcome or node in ancestors
         ]
 
-    def model_constructor(self) -> Callable[..., Callable[[], None]]:
-        """
-        Return a function that constructs the causal model defined by the Graph.
-
-        The model that is defined by a graph is a function of the parameter values,
-        or in this implementation the values of the `ParameterNode`s. This means that
-        a model can only be specified (and then used to sample from, for example) once
-        values for the `ParameterNode`s have been given. However, each of these models
-        is built in the same way, and all that needs to be done is substitute the new
-        parameter values for the old.
+    def model(self, **parameter_values: npt.ArrayLike) -> dict[str, npt.ArrayLike]:
         """
+        Model corresponding to the `Graph`'s structure.
+
+        The model created takes values of the `ParameterNode`s as keyword arguments.
```

Comment:
`self.parameter_nodes` will include all nodes where `is_parameter` is `True` - currently this is only `ParameterNode`s, but in future this could be something else. Maybe updating the docs something like this is sensible?

```suggestion
        The model created takes values of the nodes that are parameter as keyword arguments.
```